pkg/provisioner/helpers.go

```diff
@@ -209,8 +203,8 @@ func removeFinalizer(obj metav1.Object) {
 	finalizers := obj.GetFinalizers()
 	for i, f := range finalizers {
 		if f == finalizer {
-			logD.Info("removing finalizer", "object", obj.GetName(), "finalizer", finalizer)
 			obj.SetFinalizers(append(finalizers[:i], finalizers[i+1:]...))
+			logD.Info("finalizer removed")
```

Comment:
can we get the k8s object name in this msg so we see which finalizer was removed. Or does that matter?
Can `SetFinalizer` ever return an error cuz if so then I'd prefer to *only* log an error here...